### PR TITLE
feat(xml io controller)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

This PR adds reading and writing support for xml files to the io controller

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/io/xml.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    return tree
+
+
+def read_xml_stream(stream: str) -> ET.Element:
+    """Read xml formatted stream."""
+    root = ET.fromstring(stream)
+    return root
+
+
+def write_xml(root: ET.Element, file_path: Path):
```
Comment (henrikstranneheim) [inline cg/io/xml.py:23]:
```suggestion
def write_xml(root: ET.Element, file_path: Path) -> None:
```

File: tests/io/test_io_xml.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """Tests write XML."""
+    # GIVEN a xml file
+
+    # GIVEN a file path to write to
+
+    # WHEN reading the xml file
+    raw_xml_content = read_xml(file_path=xml_file_path)
+
+    # WHEN writing the xml file from dict
+    write_xml(tree=raw_xml_content, file_path=xml_temp_path)
```
Comment (henrikstranneheim) [inline tests/io/test_io_xml.py:38]: Add assert that file exists and content is unchanged

**Review by henrikstranneheim — APPROVED** [review]
Beautiful 👍

**Comment by ChrOertlin** [comment]
ignoring the xml comment since we parse trusted xml files.

**Pull request merged**